go/packages/packages.go

```diff
@@ -94,6 +96,28 @@ var run runFunc = func(cmd *exec.Cmd) ([]byte, error) {
 	return out, nil
 }
 
+func init() {
```

Comment:
Just move the *Exists checks to each file associated with that package manager.

Reply:
i think, we will end up in the same situation if we use "init" to initialize any of these values. initializing them as variables also, would not work since the variables are the first one that gets initialized during the initialization process, and then the init functions are called inside the package.

Reply:
actually, i did not want to keep these initialization in package.go. Instead, t thought that since the order of execution of init function are so specific, i would make sense to gather all into a different file "_inits.go". this way, we ensure that all the values are initialized at first and splitting them out in a separate file would also bring out this information about compilation in lexicographical ordering( it gave me real hard time yesterday while debugging the issue :( )
However, that initialization did not work at all. For some reason, the init method was not called at all.
In the interest of time, i decided to  club them together where it made most sense

Reply:
discussed offline. moved the *exists to respective package manager